tools/docker/Dockerfile

```diff
@@ -1,29 +1,58 @@
 FROM 32bit/ubuntu:16.04
 
+ENV DEBIAN_FRONTEND noninteractive
+
 # Tools
-RUN apt-get update && \
-  apt-get install -y --no-install-recommends \
-  build-essential doxygen git wget unzip python-serial python-pip \
-  default-jdk ant srecord iputils-tracepath rlwrap \
-  mosquitto mosquitto-clients gdb \
-  && apt-get clean
+RUN apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 3FA7E0328081BFF6A14DA29AA6A19B38D3D831EF && \
+  echo "deb http://download.mono-project.com/repo/ubuntu xenial main" | sudo tee /etc/apt/sources.list.d/mono-xamarin.list && \
+  apt-get -qq update && \
+  apt-get -qq -y --no-install-recommends install \
+    ant \
+    build-essential \
+    default-jdk \
+    doxygen \
+    gdb \
+    git \
+    gksu \
+    gtk-sharp2 \
+    iputils-tracepath \
+    libcanberra-gtk-module:i386 \
+    libgtk2.0-0 \
+    mono-complete \
+    mosquitto \
+    mosquitto-clients \
+    npm \
+    python-pip \
+    python-serial \
+    rlwrap \
+    screen \
+    srecord \
+    uml-utilities \
+    unzip \
+    wget \
+    > /dev/null \
```

Comment:
so, `-qq` was not enough?

Reply:
Nope, -qq will not suppress those:

```
Preconfiguring packages ...
(Reading database ... 10446 files and directories currently installed.)
Preparing to unpack .../perl-base_5.22.1-9ubuntu0.5_i386.deb ...
Unpacking perl-base (5.22.1-9ubuntu0.5) over (5.22.1-9) ...
Setting up perl-base (5.22.1-9ubuntu0.5) ...
(Reading database ... 10446 files and directories currently installed.)
Preparing to unpack .../libc6_2.23-0ubuntu10_i386.deb ...
Unpacking libc6:i386 (2.23-0ubuntu10) over (2.23-0ubuntu3) ...
Setting up libc6:i386 (2.23-0ubuntu10) ...
Processing triggers for libc-bin (2.23-0ubuntu3) ...
(Reading database ... 10446 files and directories currently installed.)
Preparing to unpack .../libssl1.0.0_1.0.2g-1ubuntu4.13_i386.deb ...
Unpacking libssl1.0.0:i386 (1.0.2g-1ubuntu4.13) over (1.0.2g-1ubuntu4.2) ...
Preparing to unpack .../python3.5_3.5.2-2ubuntu0~16.04.4_i386.deb ...
Unpacking python3.5 (3.5.2-2ubuntu0~16.04.4) over (3.5.2-2~16.01) ...
Preparing to unpack .../libpython3.5-stdlib_3.5.2-2ubuntu0~16.04.4_i386.deb ...
Unpacking libpython3.5-stdlib:i386 (3.5.2-2ubuntu0~16.04.4) over (3.5.2-2~16.01) ...
Preparing to unpack .../python3.5-minimal_3.5.2-2ubuntu0~16.04.4_i386.deb ...
Unpacking python3.5-minimal (3.5.2-2ubuntu0~16.04.4) over (3.5.2-2~16.01) ...
Preparing to unpack .../libpython3.5-minimal_3.5.2-2ubuntu0~16.04.4_i386.deb ...
Unpacking libpython3.5-minimal:i386 (3.5.2-2ubuntu0~16.04.4) over (3.5.2-2~16.01) ...
Preparing to unpack .../zlib1g_1%3a1.2.8.dfsg-2ubuntu4.1_i386.deb ...
Unpacking zlib1g:i386 (1:1.2.8.dfsg-2ubuntu4.1) over (1:1.2.8.dfsg-2ubuntu4) ...
Processing triggers for libc-bin (2.23-0ubuntu3) ...
Processing triggers for mime-support (3.59ubuntu1) ...
Setting up zlib1g:i386 (1:1.2.8.dfsg-2ubuntu4.1) ...
Processing triggers for libc-bin (2.23-0ubuntu3) ...
Selecting previously unselected package libgdbm3:i386.
```

and those generate way too much noise in the build

Reply:
ok